cpp/src/gandiva/regex_functions_holder_test.cc

```diff
@@ -650,27 +586,22 @@ TEST_F(TestExtractHolder, TestInvalidRange) {
 }
 
 TEST_F(TestExtractHolder, TestExtractInvalidPattern) {
-  std::shared_ptr<ExtractHolder> extract_holder;
-
-  auto status = ExtractHolder::Make("+", &extract_holder);
-  EXPECT_EQ(status.ok(), false) << status.message();
-
+  auto extract_holder = ExtractHolder::Make("+");
+  EXPECT_EQ(extract_holder.ok(), false) << extract_holder.status().message();
   execution_context_.Reset();
 }
 
 TEST_F(TestExtractHolder, TestErrorWhileBuildingHolder) {
-  std::shared_ptr<ExtractHolder> extract_holder;
-
   // Create function with incorrect number of params
   auto field = std::make_shared<FieldNode>(arrow::field("in", arrow::utf8()));
   auto pattern_node = std::make_shared<LiteralNode>(
       arrow::utf8(), LiteralHolder(R"((\w+) (\w+))"), false);
   auto function_node =
       FunctionNode("regexp_extract", {field, pattern_node}, arrow::utf8());
 
-  auto status = ExtractHolder::Make(function_node, &extract_holder);
-  EXPECT_EQ(status.ok(), false);
-  EXPECT_THAT(status.message(),
+  auto extract_holder = ExtractHolder::Make(function_node);
+  EXPECT_EQ(extract_holder.ok(), false);
+  EXPECT_THAT(extract_holder.status().message(),
               ::testing::HasSubstr("'extract' function requires three parameters"));
```

Comment:
Can we use `EXPECT_RAISES_WITH_MESSAGE_THAT()` here?

Reply:
Sure, updated